spec/discourse_api/api/sso_spec.rb

```diff
@@ -4,9 +4,51 @@
 describe DiscourseApi::API::SSO do
   subject { DiscourseApi::Client.new("#{host}", "test_d7fd0429940", "test_user") }
 
+  let(:params) do
+    {
+      sso_secret: 'abc',
+      sso_url: 'www.google.com',
+      name: 'Some User',
+      username: 'some_user',
+      email: 'some@email.com',
+      external_id: 'abc',
+      suppress_welcome_message: false,
+      avatar_url: 'https://www.website.com',
+      title: 'ruby',
+      avatar_force_update: false,
+      add_groups: ['a', 'b'],
+      remove_groups: ['c', 'd'],
+      # old format (which results in custom.custom.field_1 in unsigned_payload)
+      'custom.field_1' => 'tomato',
+      # new format
+      custom_fields: {
+        field_2: 'potato'
+      }
+    }
+  end
+  let(:expected_unsigned_payload) do
+    'name=Some+User&username=some_user&email=some%40email.com&'\
+    'avatar_url=https%3A%2F%2Fwww.website.com&external_id=abc&title=ruby'\
```

Comment:
note we maintain the custom.custom bug for existing users, new implementations should include `custom_fields` as a hash